src/main/java/org/wildfly/discovery/Discovery.java

```diff
@@ -190,13 +220,25 @@ public void addMatch(final ServiceURL serviceURL) {
         private final LinkedBlockingQueue<ServiceURL> queue;
         private final CopyOnWriteArrayList<Throwable> problems;
         private final DiscoveryRequest request;
+        private final long timeout;
+        private final TimeUnit timeUnit;
         private ServiceURL next;
         private boolean done;
 
         BlockingQueueServicesQueue(final LinkedBlockingQueue<ServiceURL> queue, final CopyOnWriteArrayList<Throwable> problems, final DiscoveryRequest request) {
             this.queue = queue;
             this.problems = problems;
             this.request = request;
+            this.timeout = 0;
+            this.timeUnit = TimeUnit.MILLISECONDS;
```

Comment:
`YEARS` perhaps

Reply:
My train of thought was not to block the request for too long. 0ms blocks the request for about a minute until WildFly, IIRC, cancel the connection or else it respects the timeout. Wouldn't `DAYS` block the connection or would WildFly cancel the connection in this case?

Reply:
If the blocking happens on `takeService` then the container would only block for the exact amount of time that it is willing to wait.